CMS-Enterprise/sbom-harbor
# fix: resolves remaining naming collisions and consolidates constants [ISPGCASP-1103]

## Summary

Major refactor of how constants are handled. Previously constants were declared by both `cyclonedx/` and `deploy/` modules, but recent changes to enable ephemeral environments required access to environment variables to dynamically rename resources which lead to these modules relying on each other. Ultimately all resources are named dynamically which requires access to environment variables, namely the working environment and account id. These are now passed to Lambda functions so that both cyclone/ and deploy/ can make use of `environize()`

### Added

none

### Changed

`cyclonedx/constants` contains all constants
`deploy/constants` is now just `from constants.values import *`

### Deprecated

none

### Removed

none

### Fixed

Final naming collisions

## How to test

1. Check out the branch
2. Ensure unit tests pass `python -m pytest -v --cov=cyclonedx/ tests/`
3. rename branch (locally!) to `ISPGCASP-1103<your initials>/fix-regression-that-broke-lambdas`
4. Deploy `./deploy.sh -e -u -p`
